packages/web/src/components/governance/create-proposal-modal/CreateProposalModal.tsx

```diff
@@ -0,0 +1,288 @@
+import Button from "@components/common/button/Button";
+import FormInput from "@components/common/form-input/FormInput";
+import FormTextArea from "@components/common/form-textarea/FormTextArea";
+import FormProvider from "@components/common/form/FormProvider";
+import IconAdd from "@components/common/icons/IconAdd";
+import IconClose from "@components/common/icons/IconCancel";
+import IconRemove from "@components/common/icons/IconRemove";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { DEVICE_TYPE } from "@styles/media";
+import {
+  getCreateProposalChangeParameterValidation,
+  getCreateProposalCommunityPoolSpendValidation,
+  getCreateProposalValidation,
+} from "@utils/create-proposal-validation";
+import React, {
+  useMemo,
+  useState,
+  useRef,
+  useEffect,
+  Dispatch,
+  SetStateAction,
+} from "react";
+import { useFieldArray, useForm } from "react-hook-form";
+import {
+  BoxItem,
+  CreateProposalModalBackground,
+  CreateProposalModalWrapper,
+  IconButton,
+} from "./CreateProposalModal.styles";
+
+interface Props {
+  breakpoint: DEVICE_TYPE;
+  setIsShowCreateProposal: Dispatch<SetStateAction<boolean>>;
+}
+
+interface BoxContentProps {
+  label: string;
+  children?: React.ReactNode;
+}
+
+const ProposalOption = [
+  "Text Proposal",
+  "Community Pool Spend",
+  "Parameter Change",
+];
+
+const TOKEN = {
+  urlIcon:
+    "https://raw.githubusercontent.com/Uniswap/assets/master/blockchains/ethereum/assets/0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48/logo.png",
+  currency: "GNOS",
+  value: -500,
+};
+const BoxContent: React.FC<BoxContentProps> = ({
+  label,
+  children,
+  ...props
+}) => {
+  return (
+    <BoxItem {...props}>
+      <label className="box-label">{label}</label>
+      {children}
+    </BoxItem>
+  );
+};
+
+const CreateProposalModal: React.FC<Props> = ({
+  breakpoint,
+  setIsShowCreateProposal,
+}) => {
+  const [type, setType] = useState<string>(ProposalOption[0]);
+
+  const modalRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    const closeModal = (e: MouseEvent) => {
+      if (modalRef.current && modalRef.current.contains(e.target as Node)) {
+        return;
+      } else {
+        e.stopPropagation();
+        setIsShowCreateProposal(true);
+      }
+    };
+    window.addEventListener("click", closeModal, true);
+    return () => {
+      window.removeEventListener("click", closeModal, true);
+    };
+  }, [modalRef, setIsShowCreateProposal]);
+
+  const validationProps = useMemo(() => {
+    if (type === ProposalOption[1]) {
+      return getCreateProposalCommunityPoolSpendValidation();
+    }
+    if (type === ProposalOption[2]) {
+      return getCreateProposalChangeParameterValidation();
+    }
+    return getCreateProposalValidation();
+  }, [type]);
+
+  const methods = useForm({
```

Comment:
@khiemldk 
When using `useForm`, bind the `defaultValues` type.

Check out this document.
https://react-hook-form.com/ts